riverdriver/river_driver_interface.go

```diff
@@ -32,10 +42,95 @@ import (
 type Driver[TTx any] interface {
 	// GetDBPool returns a database pool.This doesn't make sense in a world
 	// where multiple drivers are supported and is subject to change.
+	//
+	// API is not stable. DO NOT USE.
 	GetDBPool() *pgxpool.Pool
 
+	// GetExecutor gets an executor for the driver.
+	//
+	// API is not stable. DO NOT USE.
+	GetExecutor() Executor
+
+	// UnwrapExecutor gets unwraps executor from a driver transaction.
+	//
+	// API is not stable. DO NOT USE.
+	UnwrapExecutor(tx TTx) Executor
+
 	// UnwrapTx turns a generically typed transaction into a pgx.Tx for use with
 	// internal infrastructure. This doesn't make sense in a world where
 	// multiple drivers are supported and is subject to change.
+	//
+	// API is not stable. DO NOT USE.
 	UnwrapTx(tx TTx) pgx.Tx
 }
+
+// Executor provides River operations against a database. It may be a database
+// pool or transaction.
+type Executor interface {
```

Comment:
The executor will eventually contain all possible operations against a River schema. Notably, it looks quite similar to an adapter, and long term, will probably replace it.